accepted/1847 - FinalizationRegistry/proposal.md

```diff
@@ -247,15 +238,27 @@ abstract class Finalizable {
 typedef NativeFinalizer = Void Function(Pointer<Void>);
 typedef NativeFinalizerPtr = Pointer<NativeFunction<NativeFinalizer>>
 
-/// [FinalizationRegistry] which will execute its finalizers as early as
-/// possible without waiting for control to return to the event loop.
+/// [Finalizer] *may* call the given callback when objects it is attached to
+/// become inaccessible.
+///
+/// This callback is guaranteed to be called: when isolate group is shutting
+/// down for each object to which the finalizer is still attached.
 ///
-/// Will also invoke finalization callbacks when the isolate which created
-/// this finalization registry is shutting down.
-abstract class NativeFinalizationRegistry<F extends Finalizable>
-    extends FinalizationRegistry<Pointer> {
-  /// Creates a finalization registry with the given finalization
-  /// callback.
+/// No other promises are made that the callback will ever be called,
+/// only that *if* it is called with a finalization token as argument,
+/// at least one object registered in the registry with that finalization token
+/// is no longer accessible to the program.
+///
```

Comment:
Is the "no other promises" true for the native finalizer?

Reply:
We can't promise that GC ever happens. 

I have removed most of this text and replaced it with your variant which is actually much better.